### DISCUSSION TITLE
Is there a way to track the size of files uploaded to Pocketbase ?

### DISCUSSION THREAD

**Discussion by gossterrible** [pocketbase/pocketbase#6018] [Q&A] [answered]
I wanted to thank the team for PocketBase - it's been a fantastic addition to my development workflow!

I'm working on a SAAS application and need to implement a file upload quota system. My initial assumption was that file metadata (size, type, etc.) would be automatically stored with records, allowing me to query/aggregate this data to track quota usage per user.

While considering solutions, I thought about adding a pre-upload hook to compute file sizes client-side and save them as record fields. However, I'm concerned this approach could be bypassed by malicious users.

Has anyone implemented similar quota tracking for file uploads? I'd appreciate any suggestions for a more secure approach.

Thank you!

**Comment by chnfyi** [comment]
Indeed, I think you can achieve it using a custom route to update for example an _aggregatedSize_ field on each upload.

If you want to allow users to view their quota but disallow them to update the _aggregatedSize_ field themselves you can set this API rule:
`@request.body.aggregatedSize:isset = false`

**Reply by gossterrible** [reply, marked as answer]
I went ahead an implemented two hooks to help with this which I think can be extended by other https://gist.github.com/gossterrible/6009126ac7d7939c3cbd80b785b3cb25
